src/plugins/plugin.legend.js

```diff
@@ -71,17 +71,21 @@ defaults._set('global', {
 	},
 
 	legendCallback: function(chart) {
-		var text = [];
-		text.push('<ul class="' + chart.id + '-legend">');
-		for (var i = 0; i < chart.data.datasets.length; i++) {
-			text.push('<li><span style="background-color:' + chart.data.datasets[i].backgroundColor + '"></span>');
-			if (chart.data.datasets[i].label) {
-				text.push(chart.data.datasets[i].label);
+		var list = document.createElement('ul');
+		list.setAttribute('class', chart.id + '-legend');
+
+		if (datasets.length) {
```

Comment:
I'd probably remove this and instead handle it in the for-loop. Also, it's more inline with the project's coding standards to declare all variables at the top of the method. Overall this PR looks good to me though

```
var list = document.createElement('ul');
var i, ilen, listItem, listItemSpan;

list.setAttribute('class', chart.id + '-legend');

for (i = 0, ilen = datasets.length || 0; i < ilen; i++) {
```

Reply:
Note the same comment would apply to the other two files as well

Reply:
Hey @benmccan, thanks for your feedback on this. I've made additional code review fixes which you can preview in this commit: https://github.com/ian-pvd/Chart.js/pull/1/commits/1a014152e9271676154ff714b756b1aa3693e34d

Please let me know if there's anything else I can do to help progress these changes. For example, I don't seem to be able to re-open [the original PR](https://github.com/chartjs/Chart.js/pull/6179) into the chartjs repository, which I think is necessary for these changes to show up there.

Thank you again for your help here.